### PR TITLE
Components reactoring

### PR DESCRIPTION
**Pull request by ai**
- [x] Use CSS nested (it is now standard) to group rules for the same element
- [x] Simplify CSS classes
- [x] Simplify and unify component’s names
- [x] Remove some unused classes
- [x] Split huge components (`Article`, `BrowserStats`)
- [x] Add component’s prefix to every `data-id`
- [x] Separate components
- [x] Clean up HTML generation
- [x] Move from `innerHTML` to `innerText` for security
- [x] Move from `@media (max-width: 200px)` to `@media (width <= 200px)` for better readability

Closes https://github.com/browserslist/browsersl.ist/issues/316

Closes https://github.com/browserslist/browsersl.ist/issues/460

### CODE REVIEW

**Comment by ai** [comment]
I think we should not merge it today because of Product Hunt launch.

**Comment by a-chabin** [comment]
I am ready for any interpretation of BEM. 

- Block — block
- Block_element — element 
- Block_element.is-modifier — modifier (only bool, not `_size-small` / `_size-medium` etc)
Did I get this right?

Won't this frighten potential contributors?)

**Comment by ai** [comment]
> Did I get this right?

Yes. Since we don’t have two blocks on the same element, we do not need to prefix for modifier.

> Won't this frighten potential contributors?)

BEM anyway is not a common standard nowadays 😥

File: client/view/Form/Form.js [hunk 1]
```diff
@@ -47,30 +45,20 @@ function handleFormSubmit(e) {
   changeUrl(query, region)
 
   e.preventDefault()
```
Comment (a-chabin) [inline client/view/Form/Form.js:47]: oops, after some PR we have 2 `e.preventDefault` in `handleFormSubmit`

File: client/view/Form/Form.js [hunk 2]
```diff
@@ -83,7 +71,7 @@ function renderRegionSelectOptions() {
     let renderOption = (id, name) => {
       let option = document.createElement('option')
       option.value = id
```
Comment (a-chabin) [inline client/view/Form/Form.js:73]: Let's use the new `createTag` utility for better readibility

```suggestion
      let option = createTag('option', [], name)
      option.value = id
```

File: client/view/Form/Form.js [hunk 3]
```diff
@@ -83,7 +71,7 @@ function renderRegionSelectOptions() {
...
-      option.innerHTML = name
+      option.innerText = name
```
Comment (a-chabin) [inline client/view/Form/Form.js:74]:
```suggestion
```

File: client/view/Form/Form.js (hunk 1 above)
Comment (ai) [reply client/view/Form/Form.js]: Fixed e8a926b

File: client/view/Footer/Footer.css [hunk 4]
```diff
@@ -1,12 +1,7 @@
 .Footer {
   display: flex;
   align-items: center;
-  padding: 12px 32px 24px;
+  width: var(--block-width);
+  padding: var(--block-padding);
   margin: 0;
```
Comment (a-chabin) [inline client/view/Footer/Footer.css:6]: Let's make these changes:

1. Add `box-sizing: border-box`. To make the `.Footer` width equal to the `.Article`
2. Remove unused `display: flex` and `align-items: center` for `.Footer`
3. Set `display: inline-block` to `.Badge`

**Comment by a-chabin** [comment]
[Footer badge fix for this PR](https://github.com/browserslist/browsersl.ist/pull/463)

File: client/view/Footer/Footer.css (hunk 4 above)
Comment (a-chabin) [reply client/view/Footer/Footer.css:6]: Done it here
 https://github.com/browserslist/browsersl.ist/pull/463

**Review by a-chabin — APPROVED** [review]

**Pull request merged**